services/grpc/lightning.methods.js

```diff
@@ -260,73 +260,64 @@ async function closeChannel(payload = {}) {
  * sendPayment - Call lnd grpc sendPayment method.
  *
  * @param {object} payload Payload
- * @returns {Promise} SendResponse
+ * @returns {Promise} Original payload augmented with lnd sendPayment response data.
  */
 async function sendPayment(payload = {}) {
+  // Our response will always include the original payload.
+  const res = { ...payload }
+
   return new Promise((resolve, reject) => {
     try {
       const call = this.service.sendPayment(payload)
 
       call.on('data', data => {
-        const isSuccess = !data.payment_error
-        if (isSuccess) {
-          grpcLog.debug('PAYMENT SUCCESS', data)
-
-          // Convert payment_hash to hex string.
-          let paymentHash = data.payment_hash
-          if (paymentHash) {
-            paymentHash = paymentHash.toString('hex')
-          }
-
-          // In some cases lnd does not return the payment_hash. If this happens, retrieve it from the invoice.
-          else {
-            const invoice = bolt11DecodePayReq(payload.payment_request)
-            const paymentHashTag = invoice.tags
-              ? invoice.tags.find(t => t.tagName === 'payment_hash')
-              : null
-            paymentHash = paymentHashTag ? paymentHashTag.data : null
-          }
-
-          // Convert the preimage to a hex string.
-          const paymentPreimage = data.payment_preimage
-            ? data.payment_preimage.toString('hex')
+        // Convert payment_hash to hex string.
+        data.payment_hash = data.payment_hash ? data.payment_hash.toString('hex') : null
```

Comment:
Minor but could be a little cleaner if you do 
```
if (data.payment_hash) {
}
else {
}
```
so two cases are more obvious and no extract checks are performed